### PR TITLE
Feature/v1-178 Change logic of drawing buttons

### PR DESCRIPTION
**Pull request by irina-sinitsyna**
**Overview:**
> 1. Replaced links over the buttons with component={Link}
> 2. Removed additional buttons with the loader
> 3. Changed logic of drawing buttons in cases of extra rendering

### CODE REVIEW

File: frontend/src/pages/LearningCourse/CourseInfoToggle/CourseInfoToggle.tsx [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import { FC } from 'react';
 
+import { ButtonLabels } from 'constants/ButtonLabels';
+
 import { CourseInfoToggleButton, ExpandLessIcon, ExpandMoreIcon } from './styled';
 
 interface IProps {
```
Comment (PechurinaLena) [inline frontend/src/pages/LearningCourse/CourseInfoToggle/CourseInfoToggle.tsx:7]: Do we need to change Interface name sccording to component name

File: frontend/src/pages/PassingTest/TestResult/styled.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import styled from 'styled-components';
 import { Box } from '@mui/system';
```
Comment (PechurinaLena) [inline frontend/src/pages/PassingTest/TestResult/styled.ts:2]: change import styled please

Comment (irina-sinitsyna) [reply frontend/src/pages/PassingTest/TestResult/styled.ts:2]: it's already done by @ostapdans99 in v1-173

File: frontend/src/pages/LearningCourse/CourseInfoToggle/CourseInfoToggle.tsx (hunk 1 above)
Comment (irina-sinitsyna) [reply frontend/src/pages/LearningCourse/CourseInfoToggle/CourseInfoToggle.tsx:7]: I think, we'll have an additional ticket for this improvement, but I'll fix this file interface name

File: frontend/src/pages/CoursesList/AddCourseButton/AddCourseButton.tsx [hunk 3]
```diff
@@ -7,11 +7,15 @@ import { AddButton, AddButtonWrapper } from './styled';
...
-      <AddButton disableElevation variant="mediumContained">
-        {ButtonLabels.addWithPlus}
-      </AddButton>
-    </Link>
+    <AddButton
+      variant="mediumContained"
+      component={Link}
+      to={PATHS.coursesList}
+      color="primary"
+      disableElevation
```
Comment (ostapdans99) [inline frontend/src/pages/CoursesList/AddCourseButton/AddCourseButton.tsx:15]: Maybe sort props like this? 

```suggestion
disableElevation
color="primary"
variant="mediumContained"
component={Link}
to={PATHS.coursesList}
```

File: frontend/src/pages/CoursesList/CourseActions/CourseActions.tsx [hunk 4]
```diff
@@ -32,35 +32,46 @@ const CourseActions: React.FC<ICourseActions> = ({
...
-  </CourseActionsBox>
-);
+        {!isAdmin &&
+          (clientCourses ? (
+            <ActionButton
+              label={COURSE_LABELS[clientCourses[index].status]}
+              status={clientCourses[index].status}
+              progress={clientCourses[index].progress}
+              applyDate={clientCourses[index].applyDate}
+              courseId={clientCourses[index]._id}
```
Comment (ostapdans99) [inline frontend/src/pages/CoursesList/CourseActions/CourseActions.tsx:55]: Can you destructure this? 

```suggestion
const {_id, status, progress, applyDate} = clientCourses[index] || {}

label={COURSE_LABELS[status]}
status={status}
progress={progress}
applyDate={applyDate}
courseId={_id}
```

Comment (irina-sinitsyna) [reply frontend/src/pages/CoursesList/CourseActions/CourseActions.tsx:55]: @ostapdans99 this method caused eslint and ts errors such as `'Variable name _id must match one of the following formats: camelCase, PascalCase, UPPER_CASE' ` and `Type 'undefined' cannot be used as an index type.` so my solution is on the screenshoot. What do you think about this one? Should we use it or leave it without destructuring?

image

Comment (ostapdans99) [reply frontend/src/pages/CoursesList/CourseActions/CourseActions.tsx:55]: I think, more readable variant with distructure. I dont see any problems with your solution

**Review by ostapdans99 — APPROVED** [review]

**Review by PechurinaLena — APPROVED** [review]

**Review by zhuk-vezunchikz — APPROVED** [review]

**Pull request merged**